### PR TITLE
Refine stacktrace filtering

### PR DESCRIPTION
**Pull request by jandro996**
# What Does This Do
Filter out from stack classes that matches the IAST exclusion trie

# Motivation
Filter out frameworks (e.g. Spring) from vulnerability stack

# Additional Notes
Add a new value to IAST exclusion trie to manage all combinations
Keep a last resort fallback (e.g. if all frames are filtered, keep the first non-DD frame

### CODE REVIEW

**Review by manuel-alvarez-alvarez — APPROVED** [review]

File: dd-java-agent/instrumentation/iast-instrumenter/build.gradle [hunk 1]
```diff
@@ -15,8 +14,9 @@ testSets {
   }
 }
 
-compileJava.dependsOn 'generateClassNameTries'
-packageSources.dependsOn 'generateClassNameTries'
-sourcesJar.dependsOn 'generateClassNameTries'
+dependencies {
+  api project(':dd-java-agent:agent-iast')
```
Comment (smola) [inline dd-java-agent/instrumentation/iast-instrumenter/build.gradle:18]: I wonder if this interacts with how we package and load IAST classes...

Comment (smola) [reply dd-java-agent/instrumentation/iast-instrumenter/build.gradle:18]: Is this an appropriate solution?
https://github.com/DataDog/dd-trace-java/pull/3909/files#diff-e8d9d03a860ff7290549a8ba3f6f71131a278e8a8afabb00ef210e0bc17b9c9bR13-R14

Comment (jandro996) [reply dd-java-agent/instrumentation/iast-instrumenter/build.gradle]: Seems that woks on test, system test and resolve the duplication issue 👍

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java [hunk 2]
```diff
@@ -452,25 +454,20 @@ private <E> void checkInjection(
...
                 first[0] = stack;
               }
-              return IastExclusionTrie.apply(stack.getClassName()) != 1;
+              return IastExclusionTrie.apply(stack.getClassName()) < 1;
             })
         .findFirst()
         .orElse(first[0]);
   }
 
-  private static Range[] getRanges(final TaintedObject taintedObject) {
```
Comment (smola) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java:470]: Is this now unused?

**Review by smola — APPROVED** [review]

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java (hunk 2 above)
Comment (jandro996) [reply dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java:470]: yes

**Pull request merged**